config-lib.pl

```diff
@@ -407,5 +407,89 @@ sub save_module_preferences
 	}
 }
 
+# read_config_params(url, insert-prefix)
+# Parses the URL parameters from a URL, and returns a hash ref
+sub read_config_params
+{
+my ($url, $pref) = @_;
+# Extract the part of the URL after the question mark
+my ($query_string) = $url =~ /\?(.*)$/;
+my %params;
+
+# Split the query string on '&' to get the individual key-value pairs
+for my $pair (split(/&/, $query_string)) {
+	my ($key, $value) = split /=/, $pair;
+	$value = &un_urlize($value);
+	my $param_prefix = $pref ? "_cparam-" : "";
+	$params{"$param_prefix$key"} = $value;
+	}
+return \%params;
+}
+
+# print_config_posted_params()
+# Prints hidden fields for all the parameters in the referrer URL
+sub print_config_posted_params
+{
+my $env_referer = $ENV{'HTTP_REFERER'};
+my $params_referer = &read_config_params($env_referer);
+my $params_url = &read_config_params($ENV{'REQUEST_URI'});
+my $params = { %$params_referer, %$params_url };
+my @params;
+if ($params) {
+	foreach my $param (keys %{$params}) {
+		my $param_prefix = "_cparam-";
+		if ($param =~ /section|module|mode|section_next|nnext|nprev|xnavigation/ ||
```

Comment:
What are all these params that are being filtered out?

Reply:
These are standard params that are used in config page forms.

Reply:
Wouldn't it make more sense to use all the params from the referrer URL, and only params starting with _cparam- from the config page? Instead of combining them and then doing this filtering, which is fragile if another param ever gets added.

Reply:
Not really. Because we don't know the names of the params that are there in the module index before going to module's config page. The original params from index page passed in the refer are prefixed with `_cparam-` to distinguish from the standard config parameters. Moreover, later params starting with `_cparam-` has to be resubmitted across pages in the sections lister (Previous and Next buttons). 

Honestly, this is a very simple feature to have but very difficult to implement, as it involves a lot of variables, like normal link button to return to module index has to work. Forms should work when submitted and returned back to index restoring original params but also, when sections are used params needs to be passed and separated correctly, so index page doesn't receive the params in the end that weren't there at the first place.

So, no, I don't think we can simplify it in any way and have the same functionality in place.

Reply:
Probably, as I'm thinking now of it, I could separate form submits from links, and referer and requests in separate functions. But that would require a rework. It won't give any benefit in usability though.

Reply:
Maybe a better approach is for modules to have a way to pass the URL params to index.cgi into config.cgi, and then have it pass them along with the Next/Prev buttons and footer links? Relying on the referer seems tricky to me..